pyccl/core.py

```diff
@@ -762,6 +780,28 @@ def compute_linear_power(self):
         status = lib.cosmology_compute_linear_power(self.cosmo, psp, status)
         check(status, self)
 
+    def _compute_linear_power_from_arrays(self):
+        if not self._linear_power_on_input:
+            raise ValueError("Cannot compute linear power spectrum from"
+                             " input without input arrays initialized.")
+        pk_lin = Pk2D(pkfunc=None,
+                      a_arr=self.a_array,
+                      lk_arr=np.log(self.k_array),
+                      pk_arr=self.pk_array,
+                      is_logp=False,
+                      extrap_order_lok=1,
+                      extrap_order_hik=2,
+                      cosmo=None)
+
+        # needed to init some models
+        self.compute_growth()
```

Comment:
Do we really need this with the new scheme?

Reply:
You're right, I think it's no longer necessary, I will remove it.